pkg/internal/export/json/json.go

```diff
@@ -0,0 +1,239 @@
+// Package json provides an export node to print span contents as JSON objects
+package json
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+
+	"github.com/mariomac/pipes/pipe"
+	trace2 "go.opentelemetry.io/otel/trace"
+
+	"github.com/grafana/beyla/pkg/internal/request"
+)
+
+type PrintEnabled bool
+
+func (p PrintEnabled) Enabled() bool {
+	return bool(p)
+}
+
+type PrintIndent bool
+
+func PrinterNode(e PrintEnabled, indent PrintIndent) pipe.FinalProvider[[]request.Span] {
+	return func() (pipe.FinalFunc[[]request.Span], error) {
+		if !e {
+			return pipe.IgnoreFinal[[]request.Span](), nil
+		}
+
+		return jsonPrintFunc(indent)
+	}
+}
+
+type traceInfo struct {
+	TraceID      trace2.TraceID `json:"traceID"`
+	SpanID       trace2.SpanID  `json:"spanID"`
+	ParentSpanID trace2.SpanID  `json:"parentSpanID"`
+	Flags        uint8          `json:"flags,string"`
+}
+
+type Attributes map[string]string
+
+type jsonSpan struct {
```

Comment:
Is there an alternative approach we could consider? One challenge with this method is that any new or modified fields will require updates in multiple places. This approach seems suitable for a minimal representation of traces. However, if our goal is to have a comprehensive JSON translation of traces, it would be ideal to reuse an existing solution.

Reply:
@marctc I assume this relates to the toplevel discussion between yourself and @grcevski, meaning that we would be doing that as a separate PR? If not, feel free to correct me and I will investigate this one further.

Reply:
@marctc the latest commits implement that, let me know what you think.